### PR TITLE
CLOUDSTACK-9350: KVM-HA- Fix CheckOnHost for Local storage

### PR DESCRIPTION
**Pull request by abhinandanprateek**
- KVM-HA- Fix CheckOnHost for Local storage   
 - Also skip HA on VMs that are using local storage

### CODE REVIEW

**Comment by abhinandanprateek** [comment]
Marvin test output for NFS PS:

hine on Hyp 1 = 228dc462-ceab-4225-8d35-5749cbb593c2
VirtualMachine on Hyp 1 = 2caf838f-016f-4cf6-9d35-67a2e1398bb2
VirtualMachine on Hyp 1 = 2caf838f-016f-4cf6-9d35-67a2e1398bb2
====Trying SSH Connection: Host:192.168.217.22 User:root Port:22 RetryCnt:60===
===SSH to Host 192.168.217.22 port : 22 SUCCESSFUL===
{Cmd: nohup sh /root/test_host_ha.sh 400 > /dev/null 2>&1 &
via Host: 192.168.217.22} {returns: []}
====Trying SSH Connection: Host:192.168.217.24 User:root Port:22 RetryCnt:60===
====Trying SSH Connection: Host:192.168.217.24 User:root Port:22 RetryCnt:60===
===SSH to Host 192.168.217.24 port : 22 SUCCESSFUL===
===SSH to Host 192.168.217.24 port : 22 SUCCESSFUL===
{Cmd: ping -c 1 192.168.217.22 via Host: 192.168.217.24} {returns: [u'PING 192.168.217.22 (192.168.217.22) 56(84) bytes of data.', u'From 192.168.217.24 icmp_seq=1 Destination Host Unreachable', u'', u'--- 192.168.217.22 ping statistics ---', u'1 packets transmitted, 0 received, +1 errors, 100% packet loss, time 0ms', u'']}
{Cmd: ping -c 1 192.168.217.22 via Host: 192.168.217.24} {returns: [u'PING 192.168.217.22 (192.168.217.22) 56(84) bytes of data.', u'From 192.168.217.24 icmp_seq=1 Destination Host Unreachable', u'', u'--- 192.168.217.22 ping statistics ---', u'1 packets transmitted, 0 received, +1 errors, 100% packet loss, time 0ms', u'']}
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Down 
Host state is Down 
VirtualMachine on Hyp 1 = 721e1fbb-976d-4c83-b1d9-6c3fbb6b7282
VirtualMachine on Hyp 1 = 721e1fbb-976d-4c83-b1d9-6c3fbb6b7282
VirtualMachine on Hyp 1 = 3b04ea86-1357-4741-bcda-a5aec0363606
VirtualMachine on Hyp 1 = 3b04ea86-1357-4741-bcda-

**Comment by abhinandanprateek** [comment]
Marvin output with only Local storage:

49b-b201830aec39
VirtualMachine on Hyp 1 = 1d500ef3-ef56-4d11-91d8-c0e37f015236
VirtualMachine on Hyp 1 = 1d500ef3-ef56-4d11-91d8-c0e37f015236
VirtualMachine on Hyp 1 = 1d500ef3-ef56-4d11-91d8-c0e37f015236
VirtualMachine on Hyp 1 = 1d500ef3-ef56-4d11-91d8-c0e37f015236
====Trying SSH Connection: Host:192.168.217.24 User:root Port:22 RetryCnt:60===
===SSH to Host 192.168.217.24 port : 22 SUCCESSFUL===
{Cmd: nohup sh /root/test_host_ha.sh 400 > /dev/null 2>&1 &
via Host: 192.168.217.24} {returns: []}
====Trying SSH Connection: Host:192.168.217.22 User:root Port:22 RetryCnt:60===
====Trying SSH Connection: Host:192.168.217.22 User:root Port:22 RetryCnt:60===
===SSH to Host 192.168.217.22 port : 22 SUCCESSFUL===
===SSH to Host 192.168.217.22 port : 22 SUCCESSFUL===
{Cmd: ping -c 1 192.168.217.24 via Host: 192.168.217.22} {returns: [u'PING 192.168.217.24 (192.168.217.24) 56(84) bytes of data.', u'', u'--- 192.168.217.24 ping statistics ---', u'1 packets transmitted, 0 received, 100% packet loss, time 0ms', u'']}
{Cmd: ping -c 1 192.168.217.24 via Host: 192.168.217.22} {returns: [u'PING 192.168.217.24 (192.168.217.24) 56(84) bytes of data.', u'', u'--- 192.168.217.24 ping statistics ---', u'1 packets transmitted, 0 received, 100% packet loss, time 0ms', u'']}
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Up 
Host state is Alert 
Host state is Alert 
Host state is Alert 
Host state is Alert 
VirtualMachine on Hyp 1 = 6d2a0bdd-e87c-447c-8237-0c73d1a48f11
VirtualMachine on Hyp 1 = 6d2a0bdd-e87c-447c-8237-0c73d1a48f11
VirtualMachine on Hyp 1 = 6d2a0bdd-e87c-447c-8237-0c73d1a48f11
VirtualMachine on Hyp 1 = 6d2a0bdd-e87c-447c-8237-0c73d1a48f11
VirtualMachine on Hyp 1 = 07fe9109-6737-4584-979f-1abfb1c4a65a
VirtualMachine on Hyp 1 = 07fe9109-6737-4584-979f-1abfb1c4a65a
VirtualMachine on Hyp 1 = 07fe9109-6737-4584-979f-1abfb1c4a65a
VirtualMachine on Hyp 1 = 07fe9109-6737-4584-979f-1abfb1c4a65a

**Comment by jburwell** [comment]
@abhinandanprateek Jenkins failed due to no license header in the new test case, test_host_ha.py.  Travis failed because it could not find the Marvin egg.

**Comment by cristofolini** [comment]
@abhinandanprateek How about extracting lines 70-82 in `KVMInvestigator` to their own method? I think it would be nice to get a test case going for that section.

File: server/src/com/cloud/ha/HighAvailabilityManagerImpl.java [hunk 1]
```diff
@@ -264,6 +265,11 @@ public void scheduleRestartForVmsOnHost(final HostVO host, boolean investigate)
                 "Host [" + hostDesc + "] is down." + ((sb != null) ? sb.toString() : ""));
 
         for (VMInstanceVO vm : reorderedVMList) {
+            ServiceOfferingVO vmOffering = _serviceOfferingDao.findById(vm.getServiceOfferingId());
```
Comment (alexandrelimassantana) [inline server/src/com/cloud/ha/HighAvailabilityManagerImpl.java]: I would recommend putting this changed inside the if scope in line 273. The additions are related to debug and if the debug is disabled, you can skip it.

**Comment by koushik-das** [comment]
Code changes LGTM, verified changes in HighAvailabilityManagerImpl.java. Someone else needs to verify KVM related changes.

**Comment by swill** [comment]
@abhinandanprateek please rebase as this currently has merge conflicts with master.  Thanks...

**Comment by abhinandanprateek** [comment]
@swill rebased it. Thank you.

File: server/src/com/cloud/ha/HighAvailabilityManagerImpl.java (hunk 1 above)
Comment (jburwell) [inline server/src/com/cloud/ha/HighAvailabilityManagerImpl.java]: @alexandrelimassantana I don't see where ``vmOffering`` is being used only for debug.  It is being used on line 269 to determine if the VM is using local storage -- before debugging logging code.

**Comment by jburwell** [comment]
LGTM for code review

**Comment by rhtyd** [comment]
LGTM (code review)

tag:mergeready

**Comment by swill** [comment]
### CI RESULTS

```
Tests Run: 88
  Skipped: 2
   Failed: 1
   Errors: 1
 Duration: 11h 25m 09s
```

**Summary of the problem(s):**
```
ERROR: Test to verify access to loadbalancer haproxy admin stats page
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_internal_lb.py", line 854, in tearDown
    raise Exception("Cleanup failed with %s" % e)
Exception: Cleanup failed with Job failed: {jobprocstatus : 0, created : u'2016-05-07T12:50:26+0200', jobresult : {errorcode : 530, errortext : u'Failed to delete network'}, cmd : u'org.apache.cloudstack.api.command.user.network.DeleteNetworkCmd', userid : u'b90ec272-1410-11e6-9152-5254001daa61', jobstatus : 2, jobid : u'04de60c8-0aa7-4488-a076-a7475b147b47', jobresultcode : 530, jobresulttype : u'object', jobinstancetype : u'Network', accountid : u'b90e9c7d-1410-11e6-9152-5254001daa61'}
----------------------------------------------------------------------
Additional details in: /tmp/MarvinLogs/test_network_9UCT1L/results.txt
```

```
FAIL: Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Single VPC
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_internal_lb.py", line 599, in test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80
    self.execute_internallb_roundrobin_tests(vpc_offering)
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_internal_lb.py", line 668, in execute_internallb_roundrobin_tests
    self.setup_http_daemon(vm)
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_internal_lb.py", line 519, in setup_http_daemon
    self.fail("Failed to ssh into vm: %s due to %s" % (vm, e))
AssertionError: Failed to ssh into vm: <marvin.lib.base.VirtualMachine instance at 0x3624170> due to not all arguments converted during string formatting
----------------------------------------------------------------------
Additional details in: /tmp/MarvinLogs/test_network_9UCT1L/results.txt
```



**Associated Uploads**

**`/tmp/MarvinLogs/DeployDataCenter__May_07_2016_07_03_57_IONYWP:`**
* [dc_entries.obj](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/DeployDataCenter__May_07_2016_07_03_57_IONYWP/dc_entries.obj)
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/DeployDataCenter__May_07_2016_07_03_57_IONYWP/failed_plus_exceptions.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/DeployDataCenter__May_07_2016_07_03_57_IONYWP/runinfo.txt)

**`/tmp/MarvinLogs/test_host_ha_XQC3Z6:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_host_ha_XQC3Z6/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_host_ha_XQC3Z6/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_host_ha_XQC3Z6/runinfo.txt)

**`/tmp/MarvinLogs/test_network_9UCT1L:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_network_9UCT1L/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_network_9UCT1L/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_network_9UCT1L/runinfo.txt)

**`/tmp/MarvinLogs/test_vpc_routers_UTADLF:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_vpc_routers_UTADLF/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_vpc_routers_UTADLF/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1496/tmp/MarvinLogs/test_vpc_routers_UTADLF/runinfo.txt)


Uploads will be available until `2016-07-09 02:00:00 +0200 CEST`

*Comment created by [`upr comment`](https://github.com/cloudops/upr).*

**Comment by swill** [comment]
The two failures are not ones I am used to seeing in my environment, but I did a quick once through of the code and I don't think they problem is related to this PR.

I think this PR is ready to merge.  Can I get one other person to review this quickly to make sure I am not missing anything and these failures could actually be related to this PR?

@rhtyd, @jburwell, @koushik-das : CCing you guys because you guys did code review so hopefully one of you can quickly confirm that the issues I got in my CI run are not relevant.  Thx...

**Comment by koushik-das** [comment]
@swill The failures doesn't look related to this PR. This can be merged.

**Comment by swill** [comment]
Thanks for confirming @koushik-das.  👍  It is the middle of the night right now, so I am fading fast.

**Pull request merged**